api/solarperformanceinsight_api/compute.py

```diff
@@ -386,4 +394,52 @@ def run_performance_job(job: models.StoredJob, si: storage.StorageInterface):
             ),
         ]
     )
+    return monthly_energy, result_list
+
+
+def run_performance_job(job: models.StoredJob, si: storage.StorageInterface):
+    montlhy_energy, result_list = _calculate_performance(job, si)
+    save_results_to_db(job.object_id, result_list, si)
+
+
+def compare_expected_and_actual(job: models.StoredJob, si: storage.StorageInterface):
+    monthly_energy, result_list = _calculate_performance(job, si)
+    # performance granularity validation means there won't be system level
+    # perfromance and inverter level that you wouldn't want to sum
+    performance_data_ids = [
+        do.object_id
+        for do in job.data_objects
+        if do.definition.type == models.JobDataTypeEnum.actual_performance
+    ]
+    months = (
+        (job.definition.parameters.time_parameters._time_range)
+        .month.unique()  # type: ignore
+        .sort_values()
+    )
+    expected = monthly_energy["performance"]  # type: ignore
```

Comment:
Actually, running this I'm hitting a KeyError here, where monthly energy is a Series named "performance" instead of a DataFrame with a "performance" column.